### PR TITLE
Several fixes for DataSources description

### LINKED ISSUE #1975: National Data(Data source entry) cannot delete first input
This is a common issue for tables that have data source entry
The first input(e.g. number or character) cannot be deleted from Reference to data source and Comments unless deleting the row. @minotogna

### LINKED ISSUE #1976: National Data- Data sources table(validation rules):
_Reference to data source_ & _Comments_

1.Should allow deleting the first input (currently is blocked #1975)
2.Warning in case only have numbers or special characters, or a combination of both

### PR DESCRIPTION
**Pull request by sorja**
# DataSources

Resolves #1975
Resolves #1976

## Reference to data source & Comments

https://user-images.githubusercontent.com/5508251/212468978-503c1e1a-be22-4868-b557-598d28826e14.mov

## Reference to data source
Validate user input has other symbols than numbers or special characters
image

### Multiselect

https://user-images.githubusercontent.com/5508251/212636553-f15c32ec-0ccb-4b38-b1c8-ba694f289d9a.mov

Remove calculated variable

Screenshot 2023-01-16 at 10 02 29

## Year for data source

Years are same as ODP years for all tables

Screenshot 2023-01-16 at 10 57 34

### CODE REVIEW

File: src/client/components/MultiSelect/MultiSelect.tsx [hunk 1]
```diff
@@ -1,91 +1,96 @@
 import './MultiSelect.scss'
```
Comment (sorja) [inline src/client/components/MultiSelect/MultiSelect.tsx:1]: Refactor to functional component and remove ramda

File: src/client/components/MultiSelect/MultiSelect.tsx [hunk 2]
```diff
@@ -1,91 +1,96 @@
...
-                    <span className="multi-select__opened-item-label">{this.localizeOption(option)}</span>
-                  </div>
-                ),
-                (this.props as any).options
-              )}
-            </div>
-          ) : null}
-        </div>
+  // @ts-ignore
+  // @ts-ignore
```
Comment (minotogna) [inline src/client/components/MultiSelect/MultiSelect.tsx]: why this?

File: src/client/components/MultiSelect/MultiSelect.tsx [hunk 3]
```diff
@@ -1,91 +1,96 @@
...
+  return (
+    <div
```
Comment (minotogna) [inline src/client/components/MultiSelect/MultiSelect.tsx:49]: maybe it can even become a button ?

**Review by minotogna — DISMISSED** [review]

File: src/client/components/MultiSelect/MultiSelect.tsx (hunk 2 above)
Comment (sorja) [reply src/client/components/MultiSelect/MultiSelect.tsx]: Typo, removed

File: src/client/components/MultiSelect/MultiSelect.tsx (hunk 3 above)
Comment (sorja) [reply src/client/components/MultiSelect/MultiSelect.tsx:49]: Should be ok, semantics would change a bit (component left in refactorable components list)
Left out of this PR for now

**Review by minotogna — APPROVED** [review]

**Review by rfontanarosa — APPROVED** [review]

**Pull request merged**